src/ethereum/mod.rs

```diff
@@ -254,20 +275,28 @@ impl<TX> IntoBlock for Web3Block<TX> {
                 }
             },
             parent_hash: self.parent_hash,
+            uncles_hash: self.parent_hash,
+            author: self.author,
             state_root: self.state_root,
             transactions_root: self.transactions_root,
+            receipts_root: self.transactions_root,
+            logs_bloom: self.logs_bloom,
+            total_difficulty: self.total_difficulty,
             number: match self.number {
                 Some(number) => number,
                 None => {
                     return Err(Error::new(
                         ErrorKind::InvalidBlock,
                         "Block has no number".to_string(),
-                    ))
+                    ));
                 }
             },
-            gas_used: self.gas_used,
             gas_limit: self.gas_limit,
+            gas_used: self.gas_used,
             timestamp: self.timestamp,
+            extra_data: self.extra_data.clone(),
+            mix_data: self.transactions_root,
```

Comment:
can `mix_data` be "`0x`" instead of `self.transactions_root`?

Reply:
Done